### PR TITLE
Make overwrite the default behaviour for conflict resolution in addons

### PR DESCRIPTION
**Pull request by Skarlso**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

After the IPv6 discussion there was this comment from Richard: https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/3513#discussion_r917737084

There is no scenario in which you wouldn't want to overwrite, so we set it up as default behaviour.

**Special notes for your reviewer**:

**Checklist**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

### CODE REVIEW

**Comment by Ankitasw** [comment]
cc @richardcase for approval

File: pkg/cloud/services/eks/addons.go [hunk 1]
```diff
@@ -214,6 +214,6 @@ func convertConflictResolution(conflict ekscontrolplanev1.AddonResolution) *stri
 	case ekscontrolplanev1.AddonResolutionOverwrite:
 		return aws.String(eks.ResolveConflictsOverwrite)
 	default:
-		return nil
+		return aws.String(eks.ResolveConflictsOverwrite)
```
Comment (AverageMarcus) [inline pkg/cloud/services/eks/addons.go:217]: Is it worth simplifying this function?

```
func convertConflictResolution(conflict ekscontrolplanev1.AddonResolution) *string {
	if conflict == ekscontrolplanev1.AddonResolutionNone {
		return aws.String(eks.ResolveConflictsNone)
	}
	return aws.String(eks.ResolveConflictsOverwrite)
}
```

Comment (Skarlso) [reply pkg/cloud/services/eks/addons.go:217]: Sure why not. I don't think there will ever be a third option. So it makes sense to simplify it.

Comment (AverageMarcus) [reply pkg/cloud/services/eks/addons.go:217]: Yeah that's what I was thinking but wasn't 100% sure if more options could be possible. Either way, code change would be needed if they ever are so can change it back then if needed.

**Pull request merged**